nexosim-util/src/joiners.rs

```diff
@@ -0,0 +1,135 @@
+//! Thread handlers.
+//!
+//! This module contains threading related utilities.
+//!
+
+use std::fmt;
+use std::marker::PhantomData;
+use std::thread::{JoinHandle, Result as ThreadResult};
+
+use nexosim::simulation::Scheduler;
+
+type DropAction<T> = Box<dyn FnOnce(ThreadResult<T>) + Send + 'static>;
```

Comment:
instead of constraining to `'static`, can `for <'a>` be used instead, somehow?

Reply:
Current implementation cover all the use-cases that we foresee at the moment, so I would leave it as is.